lnrpc/devrpc/dev_server.go

```diff
@@ -359,12 +365,28 @@ func (s *Server) Quiesce(_ context.Context, in *QuiescenceRequest) (
 
 	op := wire.NewOutPoint(txid, in.ChanId.OutputIndex)
 	cid := lnwire.NewChanIDFromOutPoint(*op)
-	_, err = s.cfg.Switch.GetLink(cid)
+	ln, err := s.cfg.Switch.GetLink(cid)
 	if err != nil {
 		return nil, err
 	}
 
-	// TODO(proofofkeags): Add Link operation for initiating quiescence and
-	// implement the rest of this in those terms
-	return nil, fmt.Errorf("TODO(proofofkeags): Implement")
+	select {
+	case result := <-ln.InitStfu():
+		mkResp := func(b lntypes.ChannelParty) *QuiescenceResponse {
```

Comment:
Can't this just use `AndThen` to create the result for `QuiescenceResponse` then unpack once for the return value?

Reply:
I can use AndThen but this does motivate #8985 to be merged before to get the best of both approaches. What I need from that PR is `MapOk`. AndThen is not necessary here as the conversion from ChannelParty to QuiescenceResponse is not partial in that it does not need to lift to Result again:

```haskell
mkResp :: ChannelParty -> QuiescenceResponse

mapLeft :: (a -> b) -> Either a c -> Either b c

mapLeft mkResp :: Either ChannelParty c -> Either QuiescenceResponse c

type Result a = Either a Error

mapOk :: (a -> b) -> Result a -> Result b

mapOk mkResp :: Result ChannelParty -> Result QuiescenceResponse
```

Here's what it looks like with AndThen (which is an improvement, still)

```golang
mkResp := func(b lntypes.ChannelParty) fn.Result[*QuiescenceResponse] {
        return fn.Ok(&QuiescenceResponse{
                Initiator: b.IsLocal()
        }
}
return fn.AndThen(result, mkResp).Unpack()
```